idaes/tests/test_headers.py

```diff
@@ -39,11 +51,14 @@ def test_headers(package_root, patterns):
         # modify patterns to match the files that should have headers
         ff = FileFinder(package_root, glob_patterns=patterns)
         has_header, missing_header = detect_files(ff)
-        if len(missing_header) > 0:
+        # ignore empty files (probably should add option in 'detect_files' for this)
+        nonempty_missing_header = list(filter(lambda p: p.stat().st_size > 0, missing_header))
+        #
+        if len(nonempty_missing_header) > 0:
             pfx = str(package_root.resolve())
             pfx_len = len(pfx)
-            file_list = ", ".join([str(p)[pfx_len + 1:] for p in missing_header])
+            file_list = ", ".join([str(p)[pfx_len + 1:] for p in nonempty_missing_header])
```

Comment:
(nonessential coding suggestion) there is a `Path.relative_to()` method ([link](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to)) that does something similar, so this could be written e.g.

```py
file_list = ", ".join([str(p.relative_to(package_root) for p in nonempty_missing_header])
```